src/psyclone/tests/psyGen_test.py

```diff
@@ -2668,17 +2692,609 @@ def test_binaryoperation_can_be_printed():
+    assert fake_parent.symbol_table.lookup("arg4").is_input is True
+    assert fake_parent.symbol_table.lookup("arg4").is_output is True
+
+
+def test_fparser2astprocessor_parse_array_dimensions_attributes(
+        f2008_parser):
+    '''Test that process_declarations method parses multiple specifications
+    of array attributes.
+    '''
+    from fparser.common.readfortran import FortranStringReader
+    from fparser.two.Fortran2003 import Specification_Part
+    from fparser.two.Fortran2003 import Dimension_Attr_Spec
+
+    reader = FortranStringReader("dimension(:)")
+    fparser2spec = Dimension_Attr_Spec(reader)
+    shape = Fparser2ASTProcessor._parse_dimensions(fparser2spec)
+    assert shape == [None]
+
+    reader = FortranStringReader("dimension(:,:,:)")
+    fparser2spec = Dimension_Attr_Spec(reader)
+    shape = Fparser2ASTProcessor._parse_dimensions(fparser2spec)
+    assert shape == [None, None, None]
+
+    reader = FortranStringReader("dimension(3,5)")
+    fparser2spec = Dimension_Attr_Spec(reader)
+    shape = Fparser2ASTProcessor._parse_dimensions(fparser2spec)
+    assert shape == [3, 5]
+
+    reader = FortranStringReader("dimension(*)")
+    fparser2spec = Dimension_Attr_Spec(reader)
+    with pytest.raises(NotImplementedError) as error:
+        _ = Fparser2ASTProcessor._parse_dimensions(fparser2spec)
+    assert "Could not process " in str(error.value)
+    assert "Assumed-size arrays are not supported." in str(error.value)
+
+    reader = FortranStringReader("dimension(var1)")
+    fparser2spec = Dimension_Attr_Spec(reader)
+    with pytest.raises(NotImplementedError) as error:
+        _ = Fparser2ASTProcessor._parse_dimensions(fparser2spec)
+    assert "Could not process " in str(error.value)
+    assert ("Only integer literals are supported for explicit shape array"
+            " declarations.") in str(error.value)
+
+    # Test dimension and intent arguments together
+    fake_parent = KernelSchedule("dummy_schedule")
+    processor = Fparser2ASTProcessor()
+    reader = FortranStringReader("real, intent(in), dimension(:) :: array3")
+    fparser2spec = Specification_Part(reader).content[0]
+    processor.process_declarations(fake_parent, [fparser2spec], [])
+    assert fake_parent.symbol_table.lookup("array3").name == "array3"
+    assert fake_parent.symbol_table.lookup("array3").datatype == 'real'
+    assert fake_parent.symbol_table.lookup("array3").shape == [None]
+    assert fake_parent.symbol_table.lookup("array3").scope == "global_argument"
+    assert fake_parent.symbol_table.lookup("array3").is_input is True
+
+
+def test_fparser2astprocessor_parse_array_dimensions_unhandled(
+        f2008_parser, monkeypatch):
+    '''Test that process_declarations method parses multiple specifications
+    of array attributes.
+    '''
+    from fparser.common.readfortran import FortranStringReader
+    from fparser.two.Fortran2003 import Dimension_Attr_Spec
+    import fparser
+
+    def walk_ast_return(arg1, arg2):
```

Comment:
Please add a very brief docstring saying what this routine is for.

Reply:
and please do `class invalid(object)` to make pylint happier.

Reply:
Done